src/main/java/bsise/server/reply/ReplyRepository.java

```diff
@@ -25,6 +27,6 @@ public interface ReplyRepository extends JpaRepository<Reply, UUID> {
     @Query("SELECT r FROM Reply r where r.letter.user.id = :userId ORDER BY r.createdAt DESC")
     List<Reply> findTopNRepliesByUserId(UUID userId, Pageable pageable);
 
-    @Query("SELECT r FROM Reply r where r.letter.user.id = :userId ORDER BY r.createdAt DESC")
-    Page<Reply> findRepliesByOrderByCreatedAt(UUID userId, Pageable pageable);
+    @Query("SELECT r FROM Reply r where r.letter.user.id = :userId AND r.createdAt >= :startOfYear AND r.createdAt < :nextStartOfYear ORDER BY r.createdAt DESC")
+    Page<Reply> findRepliesByOrderByCreatedAt(UUID userId, LocalDateTime startOfYear, LocalDateTime nextStartOfYear, Pageable pageable);
```

Comment:
아래 제안한 코드에 따라 이렇게 변경하는걸 제안 드려요

```suggestion
    @Query("SELECT r FROM Reply r where r.letter.user.id = :userId AND r.createdAt >= :startOfYear AND r.createdAt <= :endOfYear ORDER BY r.createdAt DESC")
    Page<Reply> findRepliesByOrderByCreatedAt(UUID userId, LocalDateTime startOfYear, LocalDateTime endOfYear, Pageable pageable);
```